chemprop/utils.py

```diff
@@ -707,6 +707,12 @@ def update_prediction_args(
                 setattr(predict_args, key, override_defaults.get(key, value))
 
     # Same number of molecules must be used in training as in making predictions
+    if train_args.number_of_molecules != predict_args.number_of_molecules \
+            and not (isinstance(predict_args, FingerprintArgs) and predict_args.fingerprint_type == "MPN" and predict_args.mpn_shared and predict_args.number_of_molecules == 1):
+        raise ValueError('A different number of molecules was used in training '
+                         'model than is specified for prediction. This is only supported for models with shared MPN networks'
+                         f'and a fingerprint type of MPN. {train_args.number_of_molecules} smiles fields must be provided.')
```

Comment:
can we provide more details about this discrepancy in the error message? I.e., "You trained with X molecules as input but you're trying to predict using Y molecules as input"